File: webflux-dialogue/src/main/java/com/study/webflux/rag/application/credit/usecase/CreditQueryUseCase.java
package com.study.webflux.rag.application.credit.usecase;

import org.springframework.data.domain.Pageable;

import com.study.webflux.rag.domain.credit.model.CreditTransaction;
import com.study.webflux.rag.domain.credit.model.UserCredit;
import com.study.webflux.rag.domain.dialogue.model.UserId;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface CreditQueryUseCase {

	Mono<UserCredit> getBalance(UserId userId);

	Flux<CreditTransaction> getTransactions(UserId userId, Pageable pageable);
}

File: backend/application/src/main/kotlin/com/miyou/app/application/credit/usecase/CreditQueryUseCase.kt
package com.miyou.app.application.credit.usecase

import com.miyou.app.domain.credit.model.CreditTransaction
import com.miyou.app.domain.credit.model.UserCredit
import org.springframework.data.domain.Pageable
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface CreditQueryUseCase {
    fun getBalance(userId: String): Mono<UserCredit>

    fun getTransactions(
        userId: String,
        pageable: Pageable,
    ): Flux<CreditTransaction>
}
